firmware/port.c

```diff
@@ -707,7 +707,16 @@ void port_handle_extint(PortData *p, u32 flags) {
             if (port_pin_supports_interrupt(p, pin)) {
                 Pin sys_pin = p->port->gpio[pin];
                 if (flags & (1 << pin_extint(sys_pin))) {
-                    port_send_status(p, REPLY_ASYNC_PIN_CHANGE_N + pin);
+                    u8 response = REPLY_ASYNC_PIN_CHANGE_N + pin;
+
+                    // If the pin's state is high, set bit 3
+                    // of the response byte. This will be used to
+                    // inform "change" listeners of the present state
+                    // at the moment of the interrupt.
+                    if (pin_read(port_selected_pin(p))) {
```

Comment:
> The reason why we didn't do this on T2 is because of the chance that the pin state changes between when the interrupt is fired and the .read call occurs

This makes it so that the actual state at the moment of the interrupt is captured and no ".read" call (from JS) is necessary.